File: BackEnd_Learn/Java/auth/src/main/java/io/github/leeseojune53/auth/error/GlobalExceptionHandler.java
package io.github.leeseojune53.auth.error;

import io.github.leeseojune53.auth.error.exception.BusinessException;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@Slf4j
@RestControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(BusinessException.class)
    public ResponseEntity<ErrorResponse> handleBusinessException(BusinessException exception) {
        final var errorCode = exception.getErrorCode();
        log.warn("BusinessException : " + exception);

        return new ResponseEntity<>(new ErrorResponse(errorCode.getStatusCode(), errorCode.getMessage()),
                HttpStatus.valueOf(errorCode.getStatusCode()));
    }

}

File: BackEnd_Learn/kotlin/auth/src/main/kotlin/io/github/leeseojune53/auth/error/GlobalExceptionHandler.kt
package io.github.leeseojune53.auth.error

import io.github.leeseojune53.auth.error.exception.BusinessException
import io.github.leeseojune53.auth.error.exception.ErrorCode
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice


@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(value = [(BusinessException::class)])
    fun handleBusinessException(
        exception: BusinessException
    ) : ResponseEntity<ErrorResponse> {
        val errorCode: ErrorCode = exception.getErrorCode()

        return ResponseEntity<ErrorResponse>(
            ErrorResponse(errorCode.getStatusCode(), errorCode.getMessage()),
            HttpStatus.valueOf(errorCode.getStatusCode())
        )
    }


}